SA/design_concept/content/design/functional_requirements.typ

```diff
@@ -37,38 +37,42 @@ By default, "user" in the use case description refers to the "student user".
     caption: "Use Case Diagram"
 )<use_case_diagram>
 
+As the aim of this project is to find a visual representation of functional programming,
+the use case descriptions are kept very brief and only state the intention behind the use case but not how it must be achieved.
+
 === UC1 - Simple Function Composition
-A user wants to compose a simple function using pre-defined functions, e.g., Integer parameters.
+A user wants to execute their visually composed functions to see the effects of their functions on data.
 
 === UC2 - Function Execution
 A user wants to execute their visually composed functions.
 
 === UC3 - Recursive Function Composition
-A user wants to compose a function that is defined by itself.
+A user wants to compose a function that is defined using itself.
+For that, the user needs possibilities to distinguish between a recursive and a base case.
 
 === UC4 - Function Composition using Higher-Order Functions
-A user wants to compose functions that take other functions as their argument.
+To create generic functions that can be used in different contexts, a user wants to compose functions that take other functions as their argument.
```

Comment:
not sure whether the first sentence provides additional value. I would be fine with leaving it out

Reply:
I tried to provide the intention behind using hight-order functions

Reply:
What about:
```suggestion
To create reusable and composable functions, a user wants to compose functions that take other functions as their input, in other words, higher-order functions.
```